js/private/BUILD.bazel

```diff
@@ -156,3 +170,16 @@ copy_to_bin(
     srcs = ["js_run_devserver.mjs"],
     visibility = ["//visibility:public"],
 )
+
+write_source_files(
+    name = "js_image_layer_checked",
+    files = {
+        "js_image_layer.mjs": "//js/private/image:image.mjs",
```

Comment:
could you add some docs somewhere explaining why we need to check in a generated file rather than just execute the sources directly?